src/utils.jl

```diff
@@ -428,3 +428,33 @@ Get the amount dimension of a quantity (e.g., mol^(uamount)).
 """
 uamount(q::UnionAbstractQuantity) = uamount(dimension(q))
 uamount(d::AbstractDimensions) = d.amount
+
+"""
+Utility functions for `DynamicQuantitiesCollection`
+"""
+
+for f in (:enumerate, :length, :lastindex)
```

Comment:
I think defining maybe `eachindex` (?) would be better and would also give you `lastindex` for free

Reply:
Wait sorry you are right. But these are needed: https://docs.julialang.org/en/v1/manual/interfaces/#Indexing